### ISSUE TITLE
java.lang.IllegalArgumentException: long cannot be converted to an Element

### ISSUE THREAD

**Issue by mik9** [google/dagger#1233]
Next code cause an exception:

```java
import dagger.Component;

@Component
public interface Comp {
    @Component.Builder
    interface Builder {
        Builder someMethod(long value);
        Comp build();
    }
}
```
```
java.lang.IllegalArgumentException: long cannot be converted to an Element
```

And by this description it's hard to figure out what goes wrong. Like in example above we just forgot to add `@BindsInstance` to the method.

**Comment by ronshapiro** [comment]
@netdpb did you fix this one? I can't remember

**Comment by netdpb** [comment]
Not yet. I'll get on it.

**Comment by Eido95** [comment]
I recommend to re-open this issue, because `java.lang.IllegalArgumentException: <primitive> cannot be converted to an Element` is still thrown, with no indication about missing `@BindsInstance` annotation.

**Comment by netdpb** [comment]
@Eido95 which version of Dagger are you using? Can you post a new stacktrace, and the builder type?

**Comment by Eido95** [comment]
I'm not sure what you meant about "builder type", that's what I got:

dependencies:
```
implementation 'com.google.dagger:dagger:2.16'
implementation 'com.google.dagger:dagger-android:2.16'
annotationProcessor 'com.google.dagger:dagger-compiler:2.16'
```
stacktrace:
```
Caused by: java.lang.IllegalArgumentException: int cannot be converted to an Element
	at dagger.shaded.auto.common.MoreTypes$AsElementVisitor.defaultAction(MoreTypes.java:549)
	at dagger.shaded.auto.common.MoreTypes$AsElementVisitor.defaultAction(MoreTypes.java:544)
	at javax.lang.model.util.SimpleTypeVisitor6.visitPrimitive(SimpleTypeVisitor6.java:138)
	at com.sun.tools.javac.code.Type$JCPrimitiveType.accept(Type.java:594)
	at dagger.shaded.auto.common.MoreTypes.asElement(MoreTypes.java:541)
	at dagger.shaded.auto.common.MoreTypes.asTypeElement(MoreTypes.java:570)
	at dagger.internal.codegen.ComponentDescriptor$Factory.requirementForBuilderMethod(ComponentDescriptor.java:686)
	at dagger.internal.codegen.ComponentDescriptor$Factory.createBuilderSpec(ComponentDescriptor.java:661)
	at dagger.internal.codegen.ComponentDescriptor$Factory.create(ComponentDescriptor.java:552)
	at dagger.internal.codegen.ComponentDescriptor$Factory.forComponent(ComponentDescriptor.java:455)
	at dagger.internal.codegen.ComponentProcessingStep.process(ComponentProcessingStep.java:138)
	at dagger.internal.codegen.ComponentProcessingStep.process(ComponentProcessingStep.java:46)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:330)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:181)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
	at com.sun.tools.javac.main.Main.compile(Main.java:523)
	... 61 more

```

**Comment by netdpb** [comment]
Ah. Per https://github.com/google/dagger/releases/, I think that bug was not fixed until 2.18. Please try that release and let us know.
